lib/modules/apostrophe-images/lib/cursor.js

```diff
@@ -40,6 +40,65 @@ module.exports = {
       }
     });
 
+    self.addFilter('orientation', {
+      finalize: function() {
+        const orientation = self.get('orientation');
+        if (!orientation) {
+          return;
+        }
+
+        self.and(getCriteria(orientation));
+
+        function getCriteria(orient) {
+          if (orient === 'landscape') {
+            return {'attachment.landscape': true};
+          } else if (orient === 'portrait') {
+            return {'attachment.portrait': true};
+          }
+
+          // Last case is square
+          return {
+            'attachment.square': true
+          };
+        }
+      },
+      safeFor: 'public',
+      launder: function(s) {
+        return self.apos.launder.string(s);
+      },
+      choices: function(callback) {
+        const {choices} = options.module.filters
+          .find((filter) => filter.name === 'orientation');
+
+        return callback(null, choices);
+      }
+    });
+
+    self.addFilter('fileType', {
+      finalize: function() {
+        const fileType = self.get('fileType');
+        if (!fileType) {
+          return;
+        }
+
+        const criteria = {
+          'attachment.extension': fileType
+        };
+
+        self.and(criteria);
+      },
+      safeFor: 'public',
+      launder: function(s) {
+        return self.apos.launder.string(s);
+      },
+      choices: function(callback) {
+        const {choices} = options.module.filters
```

Comment:
How do we feel about these filters not being responsive to other filters? As written, the list of orientations would not respond to the fact that you've already filtered by tag and no images with this tag are landscape. cc @abea @stuartromanek

Reply:
We do that for "trash" and, I think, "published" because they are so fundamental, so it's not like we *never* do it.

Reply:
Your `addFilter` calls need documentation comments just before the call to `self.addFilter`, like the example below. This allows apostrophe-documentation to pick them up and include them in the documentation site:

```
    // Filter. Sets the MongoDB criteria, discarding
    // criteria previously added using this
    // method or the `and` method. For this reason,
    // `and` is a more common choice. You can also
    // pass a criteria object as the second argument
    // to any `find` method.
```

Reply:
Just added two comments to document filters.